### ISSUE TITLE
Unhandled exception: Regex match error: UTF-8 error: code points greater than 0x10ffff are not defined (ArgumentError)

### ISSUE THREAD

**Issue by fanzige** [hahwul/noir#95]
noir -b . -u https://testapp.internal.domains
              .__
  ____   ____ |__|______
 /    \ /  _ \|  \_  __ \
|   |  (  <♠️> )  ||  | \/
|___|  /\____/|__||__|
     \/                 v0.7.1

[*] Detecting technologies to base directory.
[I] Detected 16 technologies.
    php_pure
    c#-aspnet-mvc
    python_django
    python_flask
    go_echo
    go_gin
    java_armeria
    java_spring
    java_jsp
    crystal_kemal
    kotlin_spring
    oas3
    ruby_sinatra
    oas2
    ruby_rails
    raml
[*] Initiate code analysis based on the detected technology.
[*] Starting analysis of endpoints.
    17 Analyzers initialized
    Analysis to 16 technologies
Unhandled exception: Regex match error: UTF-8 error: code points greater than 0x10ffff are not defined (ArgumentError)
  from noir in '??'
  from noir in '??'
  from noir in '??'
  from noir in '??'
  from noir in '__crystal_main'
  from noir in 'main'
  from /lib/x86_64-linux-gnu/libc.so.6 in '__libc_start_main'
  from noir in '_start'
  from ???

**Comment by hahwul** [comment]
Hi @fanzige 
First, Thank you so much for submit issue! Your report is greatly appreciated! 
The issue you encountered was due to file names that couldn't be handled during the Dir.glob process. I've applied exception handling in the relevant section to address this problem. I plan to release the patch version as 0.7.2.

![](https://github.com/hahwul/noir/assets/13212227/0e5c1d16-7da4-411e-88c3-db92661b58fd)
*Looking at the number of detected techs, it seems like it was done on noir's fix directory, so I also tested it with that path.*
